build/portal/html/admin/index.php

```diff
@@ -375,6 +375,41 @@ function formatOffset($offset) {
                     </div>
                 </div>
                 <div role="tabpanel" class="tab-pane fade" id="navigation">
+                  <div class="panel panel-default">
+                      <div class="panel-heading">API Key</div>
+                      <div class="panel-body">
+                        <div class="form-group">
+                                <label for="BingMapsAPI">Bing Maps API Key</label>
+                                <input type="text" class="form-control" id="bingMapAPIKey" name="bingMapAPIKey"
+                                        value="<?php  if(!empty($_POST['bingMapAPIKey'])){
+                                        $apikey = $_POST['bingMapAPIKey'];
+                                        $path_to_file = '/usr/share/dump1090-mutability/html/config.js';
+                                        $file_contents = file_get_contents($path_to_file);
+                                        $file_contents = preg_replace('/BingMapsAPIKey = ([a-zA-Z0-9"]+)/', "BingMapsAPIKey = '" . $_POST['bingMapAPIKey'] . "'" , $file_contents);
+                                        file_put_contents($path_to_file,$file_contents);
+                                        }
+                                        else {
+                                          echo file_get_contents('/usr/share/dump1090-mutability/html/config.js');

```

Comment:
If the API key already exists, then this line will print all the JS content in the input which is not ideal.
It can be done by extracting only the api key from the JS as follows:
```php
$file = file_get_contents('/usr/share/dump1090-mutability/html/config.js');
$api = array();
preg_match("/BingMapsAPIKey = ([a-zA-Z0-9\"]+)/m", $file, $api);
echo !empty($api[1]) ? $api[1] : ''; 
```
**NOTE** this snipped is not tested